plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1511,7 +1511,18 @@ void producer_plugin::create_snapshot(producer_plugin::next_function<chain::snap
 }
 
 chain::snapshot_scheduler::snapshot_schedule_result
-producer_plugin::schedule_snapshot(const chain::snapshot_scheduler::snapshot_request_information& sri) {
+producer_plugin::schedule_snapshot(const chain::snapshot_scheduler::snapshot_request_params& srp) {
+   chain::controller& chain = my->chain_plug->chain();
+   const auto head_block_num = chain.head_block_num();
+
+   // missing start/end is set to head block num, missing end to UINT32_MAX
+   chain::snapshot_scheduler::snapshot_request_information sri = {
+      .block_spacing   = srp.block_spacing ? *srp.block_spacing : 0, 
+      .start_block_num = srp.start_block_num ? *srp.start_block_num : head_block_num + 1,
```

Comment:
Do we need the `+1` here since we compare with `start_block_num + 1` in `on_start_block`?

Reply:
Test (nodeos_snapshot_forked_test.py) starts to fail as it uses call to schedule with no parameters and compares snapshot to synchronous, create_snapshot - older call to make instant snapshot. I guess it will guarantee better that this "asap" snapshot will be executed and not just purged from queue as old. It's really should not be critical, since our goal when start block is not specified to make snapshot reasonably as fast as possible after API request.

Reply:
Sounds good, thanks Vlad. I guess if the test fails without this `+1`, it is probably needed and you are right it doesn't hurt to leave it.

Reply:
I re-tested without +1 and both tests (nodeos_snapshot_forked_test.py and nodeos_snapshot_diff_test.py) pass now. But I guess with cluster configuration and manipulation in forked test it can become more flaky and can fail once in a while, when cluster within test is slow.

Reply:
Sounds good, I say leave the +1 for safety.